### PR TITLE
Improve login form handling

### PR DESCRIPTION
**Pull request by cjmayo**
This is the login form commits from #364 without any changes. Let's get them out of the way.

### CODE REVIEW

**Comment by anarcat** [comment]
>  cjmayo requested your review on this pull request. 

By the way, you really don't need to explicitely request my review on new PRs. I seem them all land in my inbox in the GitHub web interface and in my email. This just adds an extra email in the pile and a slight irritation in the back of my mind. ;)

Now, if a PR has been sitting here waiting for a reviewer forever, it's fine to ask. It's also fine for another maintainer to just merge after a timeout, as per our policies...

File: linkchecker [hunk 1]
```diff
@@ -542,7 +542,7 @@ for entry in config["authentication"]:
         attrs["strpattern"] = attrs["pattern"].pattern
         msg = _("Enter LinkChecker password for user %(user)s" \
                 " at %(strpattern)s:") % attrs
-        entry["password"] = getpass.getpass(console.encode(msg))
+        entry["password"] = getpass.getpass(msg)
```
Comment (anarcat) [inline linkchecker:545]: shouldn't we do explicit decoding here, since it's a boundary?

Comment (mgedmin) [reply linkchecker:545]: getpass.getpass() works in Unicode:

```
mg@blynas: ~ $ python3
Python 3.8.2 (default, Mar 13 2020, 10:14:16) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import getpass
>>> getpass.getpass('Slaptažodis: ')
Slaptažodis: 
'ąčę'
```

**Comment by cjmayo** [comment]
> > cjmayo requested your review on this pull request.
> 
> By the way, you really don't need to explicitely request my review on new PRs. I seem them all land in my inbox in the GitHub web interface and in my email. This just adds an extra email in the pile and a slight irritation in the back of my mind. ;)

Noted - I'd imagined it was necessary for some automated approval step (although I guess even if so you can add yourself) and we seemed to be going for consensus.

> Now, if a PR has been sitting here waiting for a reviewer forever, it's fine to ask. It's also fine for another maintainer to just merge after a timeout, as per our policies...

Well ... there is 367, which plus this one then unblocks 366 from WIP.

I am sitting on a bunch of commits to complete the Python 2 tidyup which are going to create I guess at least half a dozen PRs... Plus a couple of interesting other PRs/Issues out there which might need to be addressed too.

**Comment by anarcat** [comment]
> Noted - I'd imagined it was necessary for some automated approval step (although I guess even if so you can add yourself) and we seemed to be going for consensus.

We do prefer to have patches reviewed and github seems to be setup to require two approvals, but i get notifications anyways, so I do get around reviewing those eventually. 

> Well ... there is 367, which plus this one then unblocks 366 from WIP.

Gotcha. See, if you wouldn't have used your "request review" credit on those, you could have notified me now. ;)

> I am sitting on a bunch of commits to complete the Python 2 tidyup which are going to create I guess at least half a dozen PRs... Plus a couple of interesting other PRs/Issues out there which might need to be addressed too.

That's awesome! Keep em coming!

**Comment by mgedmin** [comment]
I also get notifications about created PRs, and I check them several times a day, but!  when I'm not in the mood for review right then (e.g. diff is larger than my mental capacity at the time, or a PR depends on another PR and I don't feel like filtering by commits in the GH UI), I tend to dismiss the notification and forget all about it until someone posts a new comment or requests my review explicitly.

**Comment by anarcat** [comment]
same.

**Pull request merged**